src/js/jsx/sections/libraries/LibraryList.jsx

```diff
@@ -39,28 +77,47 @@ define(function (require, exports, module) {
          * @private
          * @param {string} libraryID Selected item ID
          */
-        _handleChange: function (libraryID) {
-            this.props.onLibraryChange(libraryID);
+        _handleChangeLibrary: function (libraryID) {
+            if (_LIBRARY_COMMANDS.indexOf(libraryID) === -1) {
+                this.props.onLibraryChange(libraryID);
+                return;
+            }
+
+            var selectedCommand = libraryID,
+                newLibraryName = selectedCommand === _RENAME_LIBRARY ? this.props.selected.name : "";
+
+            this.setState({
+                command: selectedCommand,
+                newLibraryName: newLibraryName
+            });
+
+            // Cancel Datalist selection
+            return false;
         },
 
         /**
          * Given the libraries, creates the datalist friendly
          * options for the library picker
          *
-         * @param {Array.<AdobeLibraryComposite>} libraries
-         *
          * @private
+         * @param {Array.<AdobeLibraryComposite>} libraries
          * @return {{title: String, id: string}}
```

Comment:
Please reflect the newly added returned keys in this JSDOC line as well (or just use object)